### DISCUSSION TITLE
JsonTypeInfo property used in two different ObjectMappers with different naming conventions

### DISCUSSION THREAD

**Discussion by hajdamak** [FasterXML/jackson-databind#4704] [Q&A]
Hi,

When using `@JsonTypeInfo` annotation `property` parameter is used specify which exact JSON field is used to determine which exact class will be used. Let's say for example this field is named `type`.  
However class that uses this annotation on one of it's fields is used by two different `ObjectMapper`s with different naming conventions set. In one naming convention this JSON field is named `type` in other `Type`.
Is it possible make `@JsonTypeInfo` be naming convention aware so right JSON field is used depending which naming convention is set on `ObjectMapper`?
If not is there any other way to achieve what `@JsonTypeInfo` does so both JSON naming conventions will work correctly?

Regards

**Comment by JooHyukKim** [comment]
There is field called `property` for `@JsonTypeInfo`.
See 

```java
@JsonTypeInfo(
    use=JsonTypeInfo.Id.CLASS, 
    include=JsonTypeInfo.As.PROPERTY,
    property="@class"
)
```

**Reply by hajdamak** [reply]
@JooHyukKim Yes, but AFAIK this property references name of the field as it is in JSON and not in class. And depending on which naming convention is used in JSON it might be named `type` or `Type`.
Are you sure that in your example `@class` refences name in class and not in JSON?

**Reply by JooHyukKim** [reply]
I see, how about `@JsonTypeIdResolver`? [Baeldung has example about it](https://www.baeldung.com/jackson-advanced-annotations)

**Reply by hajdamak** [reply]
If understand it correctly `@JsonTypeIdResolver` also uses `property` parameter that reference field name in JSON and not in class. Am I missing something?

**Reply by JooHyukKim** [reply]
No, you can override `public String idFromValueAndType(Object value, Class<?> type)` which takes both target class and JSON value.

**Comment by cowtowncoder** [comment]
There isn't anything automatic, but often `AnnotationIntrospector` has method(s) that may be overridden to customize handling of annotations -- for example by translating/changing some of the values.
So it might be worth a look to see how `JacksonAnnotationIntrospector` gets `@JsonTypeInfo` information out, perhaps it'd be possible to change observed `property` value.
It'd still need to be just one of possibilities (one for specific mapper) but might work.
